terraform/modules/application/security_groups.tf

```diff
@@ -0,0 +1,54 @@
+resource "aws_security_group" "ecs" {
+  name        = "${var.prefix}-ecs-security_group"
+  description = "ECS security group"
+  vpc_id      = var.vpc_id
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_vpc_security_group_ingress_rule" "ecs_ingress" {
+  description       = "Allow ingress on port ${var.application_port} from any IP address"
+  cidr_ipv4         = "0.0.0.0/0"
+  ip_protocol       = "tcp"
+  from_port         = var.application_port
+  to_port           = var.application_port
+  security_group_id = aws_security_group.ecs.id
+}
+
+resource "aws_vpc_security_group_egress_rule" "ecs_db_egress" {

```

Comment:
I am surprised by these two, particularly by the port specification on the ECS side (although I'm not an expert on networking) - I'd expect this to be covered by the more generic egress rules.

Reply:
So I've had more of a think - the reason I'm surprised is that in applications I've worked with before we've not restricted outbound ports (because we trust that connections initiated by our app are fine), so a generic rule has covered it. 

Restricting is not unreasonable so long as we do make sure we've covered everything.

Reply:
Agreed: happy to restrict.